libbeat/common/cli/password.go

```diff
@@ -0,0 +1,78 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cli
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+type method func(m string) (string, error)
+
+var methods = map[string]method{
+	"stdin": stdin,
+	"env":   env,
+}
+
+// ReadPassword allows to read a password passed as a command line parameter.
+// it offers several ways to read the password so it is not directly passed as a plain text argument:
+//   stdin - Will prompt the user to input the password
+//   env:VAR_NAME - Will read the password from the given env variable
+//
+func ReadPassword(def string) (string, error) {
+	if len(def) == 0 {
+		return "", errors.New("empty password definition")
+	}
+
+	var method, params string
+	parts := strings.SplitN(def, ":", 2)
+	method = parts[0]
+
+	if len(parts) == 2 {
+		params = parts[1]
+	}
+
+	m := methods[method]
+	if m == nil {
+		return "", fmt.Errorf("unknown password retrieval method %s", method)
```

Comment:
I think this will likely result in passwords inadvertently being echo'ed to the console. I think user's will expect that they can use `--password=<password>`. Rather than including the user's argument in the message you could list the allowed formats (like `unknown password source (use stdin or env:var)`).

Reply:
good point here, I would expect some users making that mistake